src/main/java/com/iota/iri/network/pipeline/ValidationStage.java

```diff
@@ -0,0 +1,97 @@
+package com.iota.iri.network.pipeline;
+
+import com.iota.iri.TransactionValidator;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.HashFactory;
+import com.iota.iri.model.TransactionHash;
+import com.iota.iri.network.FIFOCache;
+import com.iota.iri.network.neighbor.Neighbor;
+
+import static com.iota.iri.model.Hash.SIZE_IN_TRITS;
+
+/**
+ * The {@link ValidationStage} validates the given transaction, caches it as recently seen and then either submits
+ * further to the {@link ReceivedStage} or/and {@link ReplyStage} depending on whether the transaction originated from a
+ * neighbor or not.
+ */
+public class ValidationStage implements Stage {
+
+    private TransactionValidator txValidator;
+    private FIFOCache<Long, Hash> recentlySeenBytesCache;
+
+    /**
+     * Creates a new {@link ValidationStage}.
+     * 
+     * @param txValidator            the {@link TransactionValidator} to use to validate the transaction
+     * @param recentlySeenBytesCache the {@link FIFOCache} to cache the validate transaction as recently seen
+     */
+    public ValidationStage(TransactionValidator txValidator, FIFOCache<Long, Hash> recentlySeenBytesCache) {
+        this.txValidator = txValidator;
+        this.recentlySeenBytesCache = recentlySeenBytesCache;
+    }
+
+    /**
+     * Validates the transaction and caches it as 'recently seen'.
+     * 
+     * @param ctx the reply stage {@link ProcessingContext}
+     * @return a {@link ProcessingContext} either directing to only the {@link ReceivedStage} or both
+     *         {@link ReceivedStage} and {@link ReplyStage}, depending on whether the transaction came from a neighbor
+     *         or not
+     */
+    @Override
+    public ProcessingContext process(ProcessingContext ctx) {
+        ValidationPayload payload = (ValidationPayload) ctx.getPayload();
+        byte[] hashTrits = payload.getHashTrits();
+        byte[] txTrits = payload.getTxTrits();
+        Neighbor originNeighbor = payload.getOriginNeighbor();
+        Long txBytesDigest = payload.getTxBytesDigest();
+        Hash hashOfRequestedTx = payload.getHashOfRequestedTx();
+
+        // construct transaction hash and model
+        TransactionHash txHash = (TransactionHash) HashFactory.TRANSACTION.create(hashTrits, 0, SIZE_IN_TRITS);
+        TransactionViewModel tvm = new TransactionViewModel(txTrits, txHash);
+
+        try {
+            txValidator.runValidation(tvm, txValidator.getMinWeightMagnitude());
+        } catch (TransactionValidator.StaleTimestampException ex) {
+            if (originNeighbor != null) {
+                originNeighbor.getMetrics().incrSentTransactionsCount();
```

Comment:
Since I missed it before I am going to blame on a long method and my inability to concentrate on them

```suggestion
                originNeighbor.getMetrics().incrStaleTransactionsCount();
```

Reply:
Good catch! :ok_hand: